tests/components/homekit/test_type_humidifiers.py

```diff
@@ -0,0 +1,355 @@
+"""Test different accessory types: HumidifierDehumidifier."""
+from collections import namedtuple
+
+from pyhap.const import (
+    CATEGORY_HUMIDIFIER,
+    HAP_REPR_AID,
+    HAP_REPR_CHARS,
+    HAP_REPR_IID,
+    HAP_REPR_VALUE,
+)
+import pytest
+
+from homeassistant.components.homekit.const import (
+    ATTR_VALUE,
+    CONF_LINKED_HUMIDITY_SENSOR,
+    PROP_MAX_VALUE,
+    PROP_MIN_STEP,
+    PROP_MIN_VALUE,
+)
+from homeassistant.components.humidifier.const import (
+    ATTR_HUMIDITY,
+    ATTR_MAX_HUMIDITY,
+    ATTR_MIN_HUMIDITY,
+    DEFAULT_MAX_HUMIDITY,
+    DEFAULT_MIN_HUMIDITY,
+    DEVICE_CLASS_DEHUMIDIFIER,
+    DEVICE_CLASS_HUMIDIFIER,
+    DOMAIN,
+    SERVICE_SET_HUMIDITY,
+)
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_ENTITY_ID,
+    ATTR_UNIT_OF_MEASUREMENT,
+    DEVICE_CLASS_HUMIDITY,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+
+from tests.common import async_mock_service
+from tests.components.homekit.common import patch_debounce
+
+
+@pytest.fixture(scope="module")
+def cls():
+    """Patch debounce decorator during import of type_humidifiers."""
+    patcher = patch_debounce()
+    patcher.start()
+    _import = __import__(
+        "homeassistant.components.homekit.type_humidifiers",
+        fromlist=["HumidifierDehumidifier"],
+    )
+    patcher_tuple = namedtuple("Cls", ["hygrostat"])
+    yield patcher_tuple(hygrostat=_import.HumidifierDehumidifier)
+    patcher.stop()
+
+
+async def test_humidifier(hass, hk_driver, cls, events):
+    """Test if humidifier accessory and HA are updated accordingly."""
+    entity_id = "humidifier.test"
+
+    hass.states.async_set(entity_id, STATE_OFF)
+    await hass.async_block_till_done()
+    acc = cls.hygrostat(hass, hk_driver, "HumidifierDehumidifier", entity_id, 1, None)
```

Comment:
Import the device from `type_humidifiers` instead of using `cls.`

`test_type_media_players.py` has an example

Reply:
Ok, will do later today

Reply:
Not sure if the demo docs are updated or not

https://rc.home-assistant.io/integrations/demo/

Reply:
You're right, I forgot about that, will also do today

Reply:
home-assistant/home-assistant.io#13878